text/0000-variadic.md

```diff
@@ -191,7 +191,15 @@ original `VaList` as well as any clones of it. For instance, passing a `VaList`
 to a `vprintf` function as declared in this RFC semantically passes ownership
 of that `VaList`, and prevents calling the `arg` function again in the caller,
 but it remains the caller's responsibility to call `va_end`, so the compiler
-needs to insert the appropriate drop glue at the FFI boundary.
+needs to insert the appropriate drop glue at the FFI boundary. Conversely,
+functions accepting a `VaList` argument must not drop it themselves, since the
+caller will drop it instead.
+
+The C standard requires that the call to `va_end` for a `va_list` occur in the
+same function as the matching `va_start` or `va_copy` for that `va_list`. Some
```

Comment:
The standard only says

> The `va_end` macro facilitates a normal return from the function whose variable argument list was referred to by the expansion of the `va_start` macro, or the function containing the expansion of the `va_copy` macro, that initialized the `va_list` `ap`. [...]
>
> [I]f the `va_end` macro is not invoked before the return, the behavior is undefined.

which I take to mean that you're allowed to pass a `va_list *pap` to a function which will call `va_end(*pap)` for you. Although I don't imagine this ever happening at cross-module boundary, so you might get away with ignoring this possibility.